### ISSUE TITLE
Billing: cancelling bundled domain after plan was upgraded gives partial refund

### ISSUE THREAD

**Issue by hacchism** [Automattic/wp-calypso#53204]
### Steps to reproduce the behavior

1. Purchase a paid plan and register a domain by using the plan's domain credit
2. Upgrade the plan.
3. Cancel the domain within 96 hours

#### What I expected to happen

User shouldn't get any refund. It just should clear their plan's domain credit. 

#### What actually happened

User got a partial refund for the domain registration.

### Additional info

- Happened to find this while chatting with a user in 29811865-hc
- User's currency: BRL
- Payment Partner: Ebanx

**Comment by rezzap** [comment]
I wasn't able to reproduce this to get a refund while using credits or running the purchase through a credit card, but the one thing I did notice is that the domain credit did not get cleared on cancellation. 

Can you confirm the exact steps that were taken to cancel the domain so we can try to reproduce this to receive a refund @hacchism? From the chat, it sounds like the refund was processed on our end instead of by the user.

Thank you!

**Comment by hacchism** [comment]
@rezzap Thanks for looking! I couldn't reproduce the issue on my end either. So I thought this would be something related to the specific currency or payment method(or partner). 

One thing I noticed now in this specific user's case was that they upgraded to Premium first, and then to Business next. And their domain was registered at the same timing as the first Premium plan purchase(with the plan's domain credit) and was canceled after they upgraded to Business. So this specific flow might be the cause too🧐

> From the chat, it sounds like the refund was processed on our end instead of by the user.

No. The user was refunded for their domain first when they canceled only the domain on their own. And then, in the chat, I helped them cancel their plan(Business) and issued a remaining refund.

**Comment by hacchism** [comment]
> One thing I noticed now in this specific user's case was that they upgraded to Premium first, and then to Business next. And their domain was registered at the same timing as the first Premium plan purchase(with the plan's domain credit) and was canceled after they upgraded to Business. So this specific flow might be the cause too🧐

Another instance: 29836299-hc

The user followed the exact same steps as I described above. So I guess the issue is tied with this particular pattern.

**Comment by DavidRothstein** [comment]
Right, I think this does require upgrading the plan in order to trigger -- it's a known issue (see pNPgK-543-p2).

**Comment by hacchism** [comment]
Another instance: 4044623-zen

@DavidRothstein It's good to know that this was logged already there. Not sure if you would like to keep this open or not. I'd defer to you👍

**Comment by hacchism** [comment]
Another instance: 4224556-zen

**Comment by hacchism** [comment]
Another instance: 4266685-zen

**Comment by michaeldcain** [comment]
This came up in Slack recently: p1682154986316309-slack-C02BEP610

> But it's an edge case and not really something anyone's likely to bother fixing, especially since the "official policy" to get their domain credit back if they originally registered the wrong domain, rather than giving the credit back automatically) creates extra work anyway.  And similar to what Cain noticed above, this bug more or less just gives them a refund rather than giving them their domain credit back.

Given this, let's track the issue in the backlog, but aren't likely to fix it anytime soon.

**Comment by hacchism** [comment]
Another instance: 7327999-zen

**Comment by sirbrillig** [comment]
I'm going to look into this once https://github.com/Automattic/payments-shilling/issues/3173 is complete.

**Comment by michaeldcain** [comment]
From planning:

Moving this to In Progress for @sirbrillig to fix now that the main refund project has been completed.

**Comment by sirbrillig** [comment]
I updated the title and reproduction steps to clarify that an upgrade is required. Looking for the cause now. Possibly related to https://github.com/Automattic/payments-shilling/issues/2840

**Comment by sirbrillig** [comment]
I've added an automated test for this case in 170060-ghe-Automattic/wpcom and it appears to already be fixed (probably by the refactoring in https://github.com/Automattic/payments-shilling/issues/3173), so I'm going to mark this as closed.
